File: java/AuthModels.java
// Java AuthModels.java
// (full code, see integration guide for details)
package com.example.keymasterclient;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
public class AuthModels {
    @JsonInclude(JsonInclude.Include.NON_NULL)
    public static class AuthRequest {
        @JsonProperty("masterUserId")
        public String masterUserId;
        @JsonProperty("licenseKey")
        public String licenseKey;
        @JsonProperty("hwid")
        public String hwid;
        @JsonProperty("appVersion")
        public String appVersion;
        @JsonProperty("username")
        public String username;
        public AuthRequest(String masterUserId, String licenseKey, String hwid, String appVersion, String username) {
            this.masterUserId = masterUserId;
            this.licenseKey = licenseKey.toUpperCase();
            this.hwid = hwid;
            this.appVersion = appVersion;
            this.username = username;
        }
        public AuthRequest() {}
    }
    public static class AuthResponse {
        @JsonProperty("success")
        public boolean success;
        @JsonProperty("message")
        public String message;
        @JsonProperty("errorCode")
        public String errorCode;
        @JsonProperty("userStatus")
        public String userStatus;
        public AuthResponse() {}
        @Override
        public String toString() {
            return "AuthResponse{" +
                   "success=" + success +
                   ", message='" + (message != null ? message : "N/A") + '\'' +
                   ", errorCode='" + (errorCode != null ? errorCode : "N/A") + '\'' +
                   ", userStatus='" + (userStatus != null ? userStatus : "N/A") + '\'' +
                   '}';
        }
    }
}

File: .NET Latest/csharp/AuthModels.cs
namespace YourWorkspaceName // Replace with your application's actual namespace
{
    /// <summary>
    /// Represents the authentication request sent from the .NET application to the KeyMaster API.
    /// </summary>

    public class AuthRequest
    {
        /// Unique ID of the master user (usually the application owner or manager).
        
        public string MasterUserId { get; set; } = string.Empty;

        
        /// License key provided by the end-user of the application.
  
        public string LicenseKey { get; set; } = string.Empty;

        
        /// Hardware ID representing the machine running the application.
        
        public string Hwid { get; set; } = string.Empty;

        
        /// Version of the application (e.g., "1.0.0").
      
        public string AppVersion { get; set; } = string.Empty;

        
        /// Optional: Username of the end-user if the application supports individual user identities.
        
        public string? Username { get; set; }

        
        /// Optional: IP address of the client machine.
       
        public string? Ip { get; set; }
    }

    
    /// Represents the authentication response received from the KeyMaster API.
    
    public class AuthResponse
    {
        
        /// Indicates whether the authentication request was successful.
        
        public bool Success { get; set; }

        
        /// Response message providing additional information or error details.
        
        public string Message { get; set; } = string.Empty;

        
        /// Optional: Error code for failed authentication attempts.
        
        public string? ErrorCode { get; set; }

        
        /// Optional: Status of the user (e.g., active, banned).
        
        public string? UserStatus { get; set; }
    }
}
